### PR TITLE
Clearer error on modifying read-only role mappings

### PR DESCRIPTION
**Pull request by n1v0lg**
Currently, if users attempt to update a read-only role mapping through the Kibana UI, they get a generic "metadata data contains reserved fields" exception. This PR tailors the message to include a more specific failure reason if metadata contains a `_read_only` flag. 

Labelling non-issue since the code this changes is not released yet.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/PutRoleMappingRequest.java [hunk 1]
```diff
@@ -77,10 +78,19 @@ public ActionRequestValidationException validate(boolean validateMetadata) {
             validationException = addValidationError("role-mapping rules are missing", validationException);
         }
         if (validateMetadata && MetadataUtils.containsReservedMetadata(metadata)) {
-            validationException = addValidationError(
-                "metadata keys may not start with [" + MetadataUtils.RESERVED_PREFIX + "]",
-                validationException
-            );
+            if (metadata.containsKey(READ_ONLY_ROLE_MAPPING_METADATA_FLAG)) {
+                validationException = addValidationError(
+                    "metadata contains ["
```
Comment (jfreden) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/PutRoleMappingRequest.java:83]: My understanding Is this intended as a temporary fix until Kibana adds support for not allowing updates to mappings with `_read_only`? We should probably remove this special handling after that?  

If it is a temporary fix for UX purposes only, I think something like `"[" + name + "] is a read-only mapping and cannot be updated."` that way we don't expose how we determine it's a read-only mapping.

**Review by jfreden — APPROVED** [review]
LGTM!

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
Closing in favor of new copy: https://github.com/elastic/elasticsearch/pull/115951/files

**Pull request closed without merging**